From f075d4c3b126273e8de32830e40db9b523ec6d09 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Wed, 3 Apr 2019 10:58:25 +0200
Subject: [Cast MRP] Fix GCC build because of invalid default constructor

CastActivityManager::DoLaunchSessionParams constructor is set as
"noexcept" and to use the "default" implementation. But not all
the members default constructor are "noexcept" (i.e. MediaRoute).
That fails in GCC.

BUG=chromium:819294

Change-Id: Iaaf89c67f6cce83ed1d6f6f8a0b32018424f67eb

diff --git a/chrome/browser/media/router/providers/cast/cast_activity_manager.cc b/chrome/browser/media/router/providers/cast/cast_activity_manager.cc
index 4c63714ed836..854c5fc406c6 100644
--- a/chrome/browser/media/router/providers/cast/cast_activity_manager.cc
+++ b/chrome/browser/media/router/providers/cast/cast_activity_manager.cc
@@ -899,7 +899,7 @@ CastActivityManager::DoLaunchSessionParams::DoLaunchSessionParams(
       callback(std::move(callback)) {}
 
 CastActivityManager::DoLaunchSessionParams::DoLaunchSessionParams(
-    DoLaunchSessionParams&& other) noexcept = default;
+    DoLaunchSessionParams&& other) = default;
 
 CastActivityManager::DoLaunchSessionParams::~DoLaunchSessionParams() = default;
 
diff --git a/chrome/browser/media/router/providers/cast/cast_activity_manager.h b/chrome/browser/media/router/providers/cast/cast_activity_manager.h
index 70e0ede1b6ce..729e548ba327 100644
--- a/chrome/browser/media/router/providers/cast/cast_activity_manager.h
+++ b/chrome/browser/media/router/providers/cast/cast_activity_manager.h
@@ -337,7 +337,7 @@ class CastActivityManager : public cast_channel::CastMessageHandler::Observer,
         const url::Origin& origin,
         int tab_id,
         mojom::MediaRouteProvider::CreateRouteCallback callback);
-    DoLaunchSessionParams(DoLaunchSessionParams&& other) noexcept;
+    DoLaunchSessionParams(DoLaunchSessionParams&& other);
     ~DoLaunchSessionParams();
     DoLaunchSessionParams& operator=(DoLaunchSessionParams&&) = delete;
 
-- 
2.19.1

